From http://xenbits.xen.org/hg/linux-2.6.18-xen.hg/rev/731ff1f10c46
From: xen-devel@lists.xenproject.org
Patch-mainline: n/a
Subject: xencons: generalize use of add_preferred_console()
References: bnc#733022, bnc#852652

Changeset 762:a070228ac76e ("add hvc compatibility mode to xencons"
added this call just for the HVC case, without giving any reason why
HVC would be special in this regard. Use the call for all cases.

Signed-off-by: Jan Beulich <jbeulich@suse.com>

--- sle11sp3.orig/drivers/xen/console/console.c	2013-12-16 12:40:36.000000000 +0100
+++ sle11sp3/drivers/xen/console/console.c	2013-12-13 16:55:39.000000000 +0100
@@ -214,8 +214,6 @@ static int __init xen_console_init(void)
 		strcpy(kcons_info.name, "hvc");
 		if (xc_num == -1)
 			xc_num = 0;
-		if (!is_initial_xendomain())
-			add_preferred_console(kcons_info.name, xc_num, NULL);
 		break;
 
 	case XC_SERIAL:
@@ -236,6 +234,8 @@ static int __init xen_console_init(void)
 
 	wbuf = alloc_bootmem(wbuf_size);
 
+	if (!is_initial_xendomain())
+		add_preferred_console(kcons_info.name, xc_num, NULL);
 	register_console(&kcons_info);
 
  out:
